From: Debian Multimedia Maintainers <debian-multimedia@lists.debian.org>
Date: Wed, 23 Oct 2019 09:44:36 +0200
Subject: have Projucer link against the system libraries

Origin: Debian
Forwarded: no
Last-Update: 2019-08-21

since the Debian package has removed 3rd-party libraries (jpeg, png, flag,
ogg/vorbis), we must use against the system libraries.
Last-Update: 2019-08-21
---
 .../ProjectSaving/jucer_ProjectExport_Make.h       | 36 ++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h b/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h
index 88fd4f1..cdac636 100644
--- a/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h
+++ b/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h
@@ -464,6 +464,39 @@ private:
         if (isCurlEnabled() && ! isLoadCurlSymbolsLazilyEnabled())
             packages.add ("libcurl");
 
+        // on Debian, a number of 3rd-party libs have been stripped away,
+        // so we need to link to the system libraries
+        if (project.getEnabledModules().isModuleEnabled ("juce_core"))
+        {
+            packages.add ("zlib");
+        }
+        if (project.getEnabledModules().isModuleEnabled ("juce_graphics"))
+        {
+            packages.add ("libjpeg");
+            packages.add ("libpng");
+        }
+        if (project.getEnabledModules().isModuleEnabled ("juce_audio_formats"))
+        {
+            if(project.isConfigFlagEnabled ("JUCE_USE_FLAC", true))
+            {
+                packages.add ("flac");
+            }
+
+            if(project.isConfigFlagEnabled ("JUCE_USE_OGGVORBIS", true))
+            {
+                packages.add ("vorbis");
+                packages.add ("vorbisfile");
+                packages.add ("vorbisenc");
+                packages.add ("ogg");
+            }
+        }
+        if (project.getEnabledModules().isModuleEnabled ("juce_audio_devices"))
+        {
+            if(project.isConfigFlagEnabled ("JUCE_JACK", true))
+            {
+                packages.add ("jack");
+            }
+        }
         packages.removeDuplicates (false);
 
         return packages;
@@ -563,6 +596,9 @@ private:
         StringArray result (linuxLibs);
 
         auto libraries = StringArray::fromTokens (getExternalLibrariesString(), ";", "\"'");
+#ifdef DEBIAN_JUCERPROJECT_LIBS
+        libraries.addArray(StringArray::fromTokens (DEBIAN_JUCERPROJECT_LIBS, ";", "\"'"));
+#endif
         libraries.removeEmptyStrings();
 
         for (auto& lib : libraries)
